### PR TITLE
[Bug 1436421] Edit hooks in YAML instead of JSON

### PR DESCRIPTION
**Pull request by Rolikasi**
Bugzilla Bug: 1436421
replace usage of json with yaml

### CODE REVIEW

File: ui/src/components/HookForm/index.jsx [hunk 1]
```diff
@@ -67,6 +68,7 @@ const initialHook = {
     additionalProperties: false,
   },
 };
+const safeDumpOpts = { noCompatMode: true, noRefs: true };
```
Comment (helfi92) [inline ui/src/components/HookForm/index.jsx]: I think we can remove this and use default options.

File: ui/src/components/HookForm/index.jsx [hunk 2]
```diff
@@ -759,6 +760,7 @@ export default class HookForm extends Component {
                     </span>
                   </Typography>
                   <CodeEditor
+                    mode="yaml"
                     lint
                     value={taskInput}
                     onChange={this.handleTaskChange}
```
Comment (helfi92) [inline ui/src/components/HookForm/index.jsx]: For both editors, let's add (in YAML) to the title. Something like:

```jsx
<Typography variant="subtitle1">
    Task Template <small>(in YAML)</small> *
</Typography>
```

**Comment by helfi92** [comment]
> I've found a bug but I cant fix it. can you please help me @helfi92?

Hm, I wasn't able to replicate after removing the safeDump options. It should be okay.

**Comment by Rolikasi** [comment]
Thank you, @helfi92) 
Feel like superhero!)

**Pull request merged**